### PR TITLE
Add a fixit for attempting member lookup on Any.

### PR DESCRIPTION
**Pull request by rudkx**
#### What's in this pull request?

Add a fixit for attempting member lookup on Any.

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

Suggest casting to AnyObject or a more specific type.

rdar://problem/27639935

### CODE REVIEW

**Comment by CodaFi** [comment]
So cool!

I notice none of your tests check for the fixit.

**Comment by rudkx** [comment]
@CodaFi Yeah, I realized that just after pushing and am working on fixing it (or fixit-ing it), but I'm having to reverse-engineer how the fixit tests work in the progress so I'll have to get back to this in a bit.

File: test/Constraints/tuple.swift [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (CodaFi) [inline test/Constraints/tuple.swift]: Extraneous whitespace

File: lib/Sema/CSDiag.cpp [hunk 2]
```diff
@@ -2490,6 +2490,17 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
           diagnose(loc, diag::did_you_mean_raw_type);
           return; // Always prefer this over typo corrections.
         }
+      } else if (baseObjTy->isAny()) {
+        if (auto DRE = dyn_cast<DeclRefExpr>(baseExpr)) {
+          auto NameStr = DRE->getDecl()->getName().str();
```
Comment (CodaFi) [inline lib/Sema/CSDiag.cpp]: The rest of this file opts for `std::string` as scratch space.  Any reason why you're using StringRefs here?

**Comment by rudkx** [comment]
I'll re-open once I have the fixit implemented and tested properly.

File: lib/Sema/CSDiag.cpp (hunk 2 above)
Comment (rudkx) [inline lib/Sema/CSDiag.cpp]: I copied this from another diagnostic from another file. I'll take a closer look at some of the other diagnostics and update when I open the new PR.

**Pull request closed without merging**